src/query/expression/src/types/decimal.rs

```diff
@@ -542,7 +542,8 @@ impl Decimal for i256 {
     }
 
     fn from_float(value: f64) -> Self {
-        i256::from(value.to_i128().unwrap())
+        let bs = value.to_le_bytes().to_vec();
+        Self::de_binary(&mut bs.as_slice())
```

Comment:
we should rewrite the de_binary function to be  :
```
fn de_binary(bytes: &[u8]) -> Self 
```

thus we can omit to `to_vec`

Reply:
ok

Reply:
Yes, maybe we could use rewrite to `fn de_binary(bytes: &[u8]) `, we don't need the mut reference

Reply:
> Yes, maybe we could use rewrite to `fn de_binary(bytes: &[u8]) `, we don't need the mut reference

I have updated